CONTRIBUTION_GUIDE.md

```diff
@@ -0,0 +1,83 @@
+# Contribution Guide
+
+Thank you! If you've opened this guide, you're probably looking to help contribute to AI on EKS. We really appreciate
+you taking the time to help this project. To that end, we've written this guide to help you in your contributions so
+that the PR process goes by a lot smoother as well as to remind ourselves of all the things we need to keep in mind when
+we make changes to this repository.
+
+## General Contributions
+
+A few things to keep in mind when contributing anything to this repository:
+
+1) Open a discussion first: before committing code, please open a discussion so we can help guide.
+2) precommit checks: make sure to install precommit using `pre-commit install`. This will ensure that terraform is
+   linted correctly, whitespace is properly added at the end of the file, and that you are not accidentally committing
+   secrets to the repository. You may need to install some tools before precommit will run correctly. Please check
+   `.pre-commit-config.yaml` for instructions for each tool.
+
+## Contributing Core Infrastructure
+
+Infrastructure is the basic building block of AI on EKS. AI on EKS is built on the idea of a modular infrastructure: the
+infrastructure is separated into discrete components that can be combined to build a platform or illustrate an example (
+blueprint).
+
+Infrastructure is separated into 2 levels: Pre Cluster/Managed and Post Cluster
+
+1) Pre Cluster: These are the components that are required for creating a cluster and ones managed by AWS. This includes
+   the VPC, security groups, subnets, managed databases, managed observability, the cluster itself and the
+   infrastructure required within the cluster to get it to a functional state: VPC-CNI, coreDNS, storage controllers,
+   etc.
+2) Post Cluster: These are the Kubernetes resources that are deployed after the cluster is running and able to be set up
+   for AI. These are things like observability tooling, controllers for AI/ML, self-managed databases (RAG, RDBMS), etc.
+
+All infrastructure is available in `infra/base/terraform`. Pre Cluster infrastructure is created using terraform. You
+will find these as separate `.tf` files. If you would like to contribute new Pre Cluster infrastructure, please create a
+new terraform module for it.
```

Comment:
replace `terraform module` with `terraform resources`. We want to leverage OSS TF modules as much as possible than building our own and maintain it.